app/api/tickets.py

```diff
@@ -240,12 +240,12 @@ def before_update_object(self, ticket, data, view_kwargs):
                     {'event_id': ticket.event.id},
                     "Event having paid ticket must have a payment method",
                 )
-        else:
-            if (data.get('deleted_at') and ticket.has_current_orders):
-                raise ForbiddenError(
-                    {'source': ''},
-                    "Can't delete a ticket that has sales",
-                )
+        
+        if (ticket.has_current_orders and data.get('deleted_at')):
```

Comment:
deleted_at should be checked first. There is no need to make an SQL query everytime there is a change in tickets. It only needs to be checked if ticket is being deleted